shell/platform/darwin/ios/framework/Source/platform_message_router.mm

```diff
@@ -22,12 +64,12 @@
     FlutterBinaryMessageHandler handler = it->second;
     NSData* data = nil;
     if (message->hasData()) {
-      data = GetNSDataFromVector(message->data());
+      data = [FlutterMessageData dataWithMessage:std::move(message)];
```

Comment:
It seems like it would be much safer to change `PlatformMessage` to allow for extracting the data with ownership, so we could actually just construct a no-copy NSData here, rather than using a proxy.

Reply:
Yea, you're probably right.  I'll have to investigate, they are using shared pointers for the platformmessage, but I think they are essentially unique_ptr's.  If they are truly unique_ptr's we can steal ownership of the buffer.  It might help us on the android side too.  I can look into this today